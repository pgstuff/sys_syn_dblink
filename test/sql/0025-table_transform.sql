BEGIN;

CREATE EXTENSION sys_syn_dblink;

CREATE SCHEMA processor_data
        AUTHORIZATION postgres;

CREATE SCHEMA put
        AUTHORIZATION postgres;

CREATE SCHEMA sys_syn_proc
        AUTHORIZATION postgres;

SELECT  dblink_connect('sys_syn_test', 'dbname=contrib_regression host=' ||
        quote_literal(split_part((SELECT pg_settings.setting FROM pg_settings WHERE pg_settings.name = 'unix_socket_directories'), ', ', 1)));
SELECT dblink_exec('sys_syn_test', 'BEGIN');

INSERT INTO sys_syn_dblink.in_groups_def VALUES ('sys_syn_test', 'in');
INSERT INTO sys_syn_dblink.out_groups_def VALUES ('sys_syn_test', 'out');
INSERT INTO sys_syn_dblink.put_groups_def VALUES ('put');

INSERT INTO sys_syn_dblink.put_table_transforms (
        priority,       put_group_id_like,      new_put_schema
) VALUES (
        150,            'put',                  'put'
);

INSERT INTO sys_syn_dblink.put_table_transforms (
        priority,       dblink_connname_like,   in_group_id_like,       new_proc_schema
) VALUES (
        160,            'sys_syn_test',         'in',                   'sys_syn_proc'
);

SELECT sys_syn_dblink.processing_table_create (
        proc_schema     => 'processor_data',
        in_table_id     => 'test_table',
        out_group_id    => 'out',
        put_group_id    => 'put',
        dblink_connname => 'sys_syn_test');


SELECT * FROM sys_syn_proc.test_table_out_claim();

SELECT * FROM sys_syn_proc.test_table_out_pull();

SELECT  trans_id_in, delta_type, queue_priority, hold_updated, prior_hold_reason_count, prior_hold_reason_id, prior_hold_reason_text, id, attributes, no_diff
FROM    sys_syn_proc.test_table_out_processing
ORDER BY id, attributes;

SELECT * FROM sys_syn_proc.test_table_out_process();

SELECT  *
FROM    put.test_table_out
ORDER BY test_table_id, test_table_text;

SELECT  hold_reason_id, hold_reason_text, queue_priority
FROM    sys_syn_proc.test_table_out_processed
ORDER BY id;



SELECT dblink_exec('sys_syn_test', 'ROLLBACK');
SELECT dblink_disconnect('sys_syn_test');

ROLLBACK;
