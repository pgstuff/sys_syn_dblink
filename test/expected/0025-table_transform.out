BEGIN;
CREATE EXTENSION sys_syn_dblink;
CREATE SCHEMA processor_data
        AUTHORIZATION postgres;
CREATE SCHEMA put
        AUTHORIZATION postgres;
CREATE SCHEMA sys_syn_proc
        AUTHORIZATION postgres;
SELECT  dblink_connect('sys_syn_test', 'dbname=contrib_regression host=' ||
        quote_literal(split_part((SELECT pg_settings.setting FROM pg_settings WHERE pg_settings.name = 'unix_socket_directories'), ', ', 1)));
 dblink_connect 
----------------
 OK
(1 row)

SELECT dblink_exec('sys_syn_test', 'BEGIN');
 dblink_exec 
-------------
 BEGIN
(1 row)

INSERT INTO sys_syn_dblink.in_groups_def VALUES ('sys_syn_dblink-test', 'in');
INSERT INTO sys_syn_dblink.out_groups_def VALUES ('sys_syn_dblink-test', 'out');
INSERT INTO sys_syn_dblink.put_groups_def VALUES ('put');
INSERT INTO sys_syn_dblink.put_table_transforms (
        priority,       put_group_id_like,      new_put_schema
) VALUES (
        150,            'put',                  'put'
);
INSERT INTO sys_syn_dblink.put_table_transforms (
        priority,       dblink_connname_like,   in_group_id_like,       new_proc_schema
) VALUES (
        160,            'sys_syn_test',         'in',                   'sys_syn_proc'
);
SELECT sys_syn_dblink.proc_table_create (
        proc_schema     => 'processor_data',
        in_table_id     => 'test_table',
        out_group_id    => 'out',
        put_group_id    => 'put',
        dblink_connname => 'sys_syn_test');
 proc_table_create 
-------------------
 
(1 row)

SELECT * FROM sys_syn_proc.test_table_claim_1();
 test_table_claim_1 
--------------------
 t
(1 row)

SELECT * FROM sys_syn_proc.test_table_pull_1();
 test_table_pull_1 
-------------------
 t
(1 row)

SELECT  trans_id_in, delta_type, queue_priority, hold_updated, prior_hold_reason_count, prior_hold_reason_id, prior_hold_reason_text, id, attributes, no_diff
FROM    sys_syn_proc.test_table_processing_1
ORDER BY id, attributes;
 trans_id_in | delta_type | queue_priority | hold_updated | prior_hold_reason_count | prior_hold_reason_id | prior_hold_reason_text | id  |   attributes    | no_diff 
-------------+------------+----------------+--------------+-------------------------+----------------------+------------------------+-----+-----------------+---------
           1 | Add        |                |              |                         |                      |                        | (1) | ("test_data 1") | 
           1 | Add        |                |              |                         |                      |                        | (2) | ("test_data 2") | 
           1 | Add        |                |              |                         |                      |                        | (3) | ("test_data 3") | 
(3 rows)

SELECT * FROM sys_syn_proc.test_table_process_1();
 test_table_process_1 
----------------------
 t
(1 row)

SELECT  *
FROM    put.test_table
ORDER BY test_table_id, test_table_text;
 test_table_id | test_table_text 
---------------+-----------------
             1 | test_data 1
             2 | test_data 2
             3 | test_data 3
(3 rows)

SELECT  hold_reason_id, hold_reason_text, queue_priority
FROM    sys_syn_proc.test_table_processed_1
ORDER BY id;
 hold_reason_id | hold_reason_text | queue_priority 
----------------+------------------+----------------
                |                  |               
                |                  |               
                |                  |               
(3 rows)

SELECT dblink_exec('sys_syn_test', 'ROLLBACK');
 dblink_exec 
-------------
 ROLLBACK
(1 row)

SELECT dblink_disconnect('sys_syn_test');
 dblink_disconnect 
-------------------
 OK
(1 row)

ROLLBACK;
